chapters/en/slides/module7/module7_04.Rmd

```diff
@@ -0,0 +1,342 @@
+and you will see that it renders correctly.
+We included a link for this in the slide that you can try.
+
+This is the same that happens if we click the "Open in Vega Editor" entry
+in the three dot action button that is automatically included in each Altair chart.
+
+If you look at the HTML file we saved in the previous step
+you will see that this JSON spec is included together with some additional code
+that tells your browser to use Vega-Lite's web libraries to render this JSON spec.
+
+---
+
+## Charts can be rendered offline
+
+
+- [Link to install the offline Vega-Lite extension in JupyterLab](https://altair-viz.github.io/user_guide/display_frontends.html#displaying-in-jupyterlab)
+- Code to include a fallback image representation of your chart for offline rendering:
+
+    ```
+    alt.renderers.enable('mimetype')
+    ```
+
+Notes:
+Since displaying the JSON spec correctly depends on a the Vega-Lite web libraries,
+this means that our charts will only render if we have an internet connection.
+
+If you are working offline and want to render charts in JupyterLab,
+there are a few different options.
+
+If you want interactive chart elements to work,
+you need to install a JupyterLab extension
+that downloads the the Vega-Lite rendering library to your machine.
+[The steps for installing this extension are outlined in the Altair documentation](https://altair-viz.github.io/user_guide/display_frontends.html#displaying-in-jupyterlab)
+
+If you are OK with seeing a static version of your charts when you are offline,
+you could instruct Altair to use the Vega-Lite libraries
+only if there is an active internet connection
+and otherwise use and image for the plot.
+To do this,
+you can include the code from this slide on top of your notebook,
+just after your import statements.
+
+---
+
+## What if we want to save our charts as images instead?
+
+- The three dots action buttong can be used to save images manually
+- `altair_saver` can be used to save charts programatically
+    - [Instructions for how to install and work with Altair saver can be found here](https://github.com/altair-viz/altair_saver/).
+
+Notes:
+If we want to save a chart as a PNG image file
+(or an SVG vector text file similar to a PDF),
+we can use the three dot action button
+as we saw in the very first module of this course.
+When we are clicking the "Save as PNG" entry in this menu,
+the browser converts the Vega-Lite spec into a PNG image
+that is downloaded to our computer.
+
+If we want to save an image as a PNG programatically
+we need to use a helper package called `altair_saver`.
+This package let's Altair access the save functionality of the browser
+in a similar way as when you click the thee dot action button.
+
+[Instructions for how to install
+and work with Altair saver can be found here](https://github.com/altair-viz/altair_saver/).
+After it is installed,
+you can save as PNG by typing `chart.save('my-chart.png')`.
+
```

Comment:
I think we need a slide on which type of image to save things as (i.e., what is the functional difference between png and svg, for example). We have a section on this in Data Science: A First Introduction (DSCI 100 textbook) that you can use as inspiration: https://ubc-dsci.github.io/introduction-to-datascience/viz.html#saving-the-visualization (note: I think it's licensed as Non-Commercial, so you will need to paraphrase)